### DISCUSSION TITLE
Request Status code

### DISCUSSION THREAD

**Discussion by lupy100** [frontendbr/forum#1090] [Perguntas e Respostas]
Boa tarde pessoal, tudo bem? Estou tendo alguns problemas para entender como funciona a parte de como pegar o status code na minha request e onde seria o melhor lugar para tratar isso, estou utilizando o redux também, obrigado pela atenção

**Comment by victorsilent** [comment]
Bom, olhando alguns exemplos aqui no forum e tambem projetos de outras pessoas, o que costumo fazer é tratar tudo isso nas actions, no store do redux o ideal é tratar só a adição dos dados.
O status tu pega dentro da action tambem.
![image](https://user-images.githubusercontent.com/8001358/40209278-afcb728c-5a14-11e8-87a7-7c49aa250f9a.png)

**Comment by ghost** [comment]
@lupy100 normalmente eu resolvo com uma class que trata erro. dai com axios uso o intercepto dele. Um exemplo simples:

```
const createAxios = () => {
  const axiosInstance = nativeAxios.create({
    baseURL: BASE_URL
  })

  axiosInstance.defaults.headers.common.Authorization = getFullToken()

  axiosInstance.interceptors.response.use(
    response => response.data,
    error => {
      if (!error.response) {
        throw new Error(ERROR_TIMEOUT)
      }

      if (error.response.status === HTTP_STATUS.UNAUTHORIZED) {
        destroyToken()
      }

      return Promise.reject(error)
    }
  )
  return axiosInstance
}


```

Dá pra criar uma class só pra tratar erros tipo ErrorHandler, usar uns hashmaps e fica sucesso. Esse é um exemplo o mais didático que consegui criar aqui, dá pra disparar uns sagas, da pra disparar actions, da pra combinar com observable, enfim ai sua mente é seu guia hehehe.
